src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -96,21 +96,22 @@ private BoundExpression BindSwitchExpression(SwitchExpressionSyntax node, Diagno
             PatternSyntax node,
             TypeSymbol inputType,
             uint inputValEscape,
+            bool permitDesignations,
             bool hasErrors,
             DiagnosticBag diagnostics)
         {
             return node switch
             {
                 DiscardPatternSyntax p => BindDiscardPattern(p, inputType),
-                DeclarationPatternSyntax p => BindDeclarationPattern(p, inputType, inputValEscape, hasErrors, diagnostics),
+                DeclarationPatternSyntax p => BindDeclarationPattern(p, inputType, inputValEscape, permitDesignations, hasErrors, diagnostics),
```

Comment:
Not essential, but it feels like declaring a struct type and passing it around could reduce some repetition when it becomes necessary to pass more context down the tree when binding a pattern.

Reply:
I will look at refactoring it in the future.